lib/date_time/date_time.c

```diff
@@ -64,8 +68,13 @@ static struct sntp_time sntp_time;
 K_SEM_DEFINE(time_fetch_sem, 0, 1);
 
 static struct k_work_delayable time_work;
+#if defined(CONFIG_DATE_TIME_AUTO_UPDATE)
+static void date_time_init_auto_work(struct k_work *work);
```

Comment:
I think they can be combined. You could just remove `date_time_init_auto_work` and just use `time_work` whenever you want to trigger an update.

_The problem currently is the way they are scheduled. time_work is scheduled after the interval. And date_time_init_auto_work is scheduled when LTE is connected._

Is this really an issue? I don't see any immediate conflicts of just scheduling time_work on those occasions. Either way, time will get updated.

Reply:
Sounds good, just have to make sure that the work is only scheduled once at a time.

Reply:
This has been now changed and now I'm thinking why do we even need to trigger time update after getting to LTE network because we get XTIME notification after its subscription was moved to SEARCHING state.

Reply:
This is still needed for NTP if we don't have network time at all